packages/evm/contracts/interfaces/ICyphernodeRegistry.sol

```diff
@@ -4,48 +4,40 @@ pragma solidity >=0.8.26;
 interface ICyphernodeRegistry {
     /// @notice This event MUST be emitted when a committee is selected for an E3.
     /// @param e3Id ID of the E3 for which the committee was selected.
-    /// @param pools Addresses of the pools of nodes from which the committee was selected.
+    /// @param filter Address of the contract that will coordinate committee selection.
     /// @param threshold The M/N threshold for the committee.
     event CommitteeRequested(
         uint256 indexed e3Id,
-        address[] pools,
+        address filter,
         uint32[2] threshold
     );
 
     /// @notice This event MUST be emitted when a committee is selected for an E3.
     /// @param e3Id ID of the E3 for which the committee was selected.
-    /// @param nodes Addresses of the nodes in the committee.
-    /// @param merkleRoots Merkle roots of the nodes in the committee.
     /// @param publicKey Public key of the committee.
-    event CommitteeSelected(
-        uint256 indexed e3Id,
-        address[] nodes,
-        bytes32[] merkleRoots,
-        bytes publicKey
-    );
+    event CommitteeSelected(uint256 indexed e3Id, bytes publicKey);
 
-    /// @notice This event MUST be emitted when a node is added to the registry.
-    /// @param nodeId ID of the node.
-    /// @param node Address of the node.
-    event NodeAdded(uint256 indexed nodeId, address indexed node);
+    /// @notice This event MUST be emitted when a filter is added to the registry.
+    event FilterAdded(address indexed filter);
 
-    /// @notice This event MUST be emitted when a node is removed from the registry.
-    /// @param nodeId ID of the node.
-    /// @param node Address of the node.
-    event NodeRemoved(uint256 indexed nodeId, address indexed node);
+    /// @notice This event MUST be emitted when a filter is removed from the registry.
+    event FilterRemoved(address indexed filter);
```

Comment:
I wonder if filter removal is necessary?
Given that filters are passed in by the requestor, it seems unnessary to need to explicityly remove them. Requestors could simply stop requesting them.

Reply:
Right, agreed. It crossed my mind, and I kept it in because I thought it might be useful for the registry to exclude some filters in an emergency situation. I will remove it.

Reply:
Yeah, I was wondering about this as well.

My worry here is that it implies an untenable governance risk.